src/cartridge.rs

```diff
@@ -0,0 +1,78 @@
+use std::{
+    fs::File,
+    io::{self, Read},
+    rc::Rc,
+};
+
+///Contains all the information about the header of the cartridge
+pub(crate) struct CartridgeHeader {
+    title: String,
+    //WIP, missing properties
+}
+
+impl CartridgeHeader {
+    pub fn new(data: &[u8]) -> Self {
+        Self::verify_checksum(data).expect("Invalid checksum");
+        let title = String::from_utf8(data[0x00A0..0x00AC].to_vec()).expect("Invalid UTF-8 title");
+        if !title.is_ascii() {
+            panic!("Expected an ascii sequence");
+        }
+
+        /*fetch other data here*/
```

Comment:
Inconsistency in comment format.

It would be better to use the format
// Comment
or for docs
/// Comment

With a space between the slashes and the first word of the comment, and the first letter capitalized.